dockerfiles/init/modules/openshift/files/scripts/ocp.sh

```diff
@@ -0,0 +1,291 @@
+
+DEFAULT_CONFIG_DIR="/tmp/config"
+export CONFIG_DIR=${CONFIG_DIR:-${DEFAULT_CONFIG_DIR}}
+}
+
+get_tools() {
+    TOOLS_DIR="/tmp"
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        OC_PACKAGE="openshift-origin-client-tools-v3.6.0-c4dd4cf-mac.zip"
+        JQ_PACKAGE="jq-osx-amd64"
+        ARCH="unzip -d $TOOLS_DIR"
+        EXTRA_ARGS=""
+    else
+        OC_PACKAGE="openshift-origin-client-tools-v3.6.0-c4dd4cf-linux-64bit.tar.gz"
+        JQ_PACKAGE="jq-linux64"
+        ARCH="tar --strip 1 -xzf"
+        EXTRA_ARGS="-C $TOOLS_DIR"
+    fi
+    OC_URL=https://github.com/openshift/origin/releases/download/v3.6.0/$OC_PACKAGE
+    JQ_URL=https://github.com/stedolan/jq/releases/download/jq-1.5/$JQ_PACKAGE
+    OC_BINARY="$TOOLS_DIR/oc"
+    JQ_BINARY="$TOOLS_DIR/jq"
+
+    if [ ! -f $OC_BINARY ]; then
+        echo "download oc client..."
+        wget -q -O $TOOLS_DIR/$OC_PACKAGE $OC_URL
+        eval "$ARCH" "$TOOLS_DIR"/"$OC_PACKAGE" "$EXTRA_ARGS" &>/dev/null
+        rm -rf "$TOOLS_DIR"/README.md "$TOOLS_DIR"/LICENSE "${TOOLS_DIR:-/tmp}"/"$OC_PACKAGE"
+    fi
+
+    if [ ! -f $JQ_BINARY ]; then
+        echo "download jq..."
+        wget -q -O $JQ_BINARY $JQ_URL
+        chmod +x $JQ_BINARY
+    fi
+    export PATH=${PATH}:${TOOLS_DIR}
+}
+
+ocp_is_booted() {
+    # for now we check only openshift registry because this is enough
+    ocp_registry_container_id=$(docker ps -a  | grep openshift/origin-docker-registry | cut -d ' ' -f1)
+    if [ ! -z "$ocp_registry_container_id" ];then
+        ocp_registry_container_status=$(docker inspect "$ocp_registry_container_id" | $JQ_BINARY .[0] | $JQ_BINARY -r '.State.Status')
+    else
+        return 1
+    fi
+    if [[ "${ocp_registry_container_status}" == "running" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+wait_ocp() {
+  OCP_BOOT_TIMEOUT=120
+  echo "[OCP] wait for ocp full boot..."
+  ELAPSED=0
+  until ocp_is_booted; do
+    if [ ${ELAPSED} -eq "${OCP_BOOT_TIMEOUT}" ];then
+        echo "OCP didn't started in $OCP_BOOT_TIMEOUT secs, exit"
+        exit 1
+    fi
+    sleep 2
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+run_ocp() {
+    $OC_BINARY cluster up --public-hostname="${OC_PUBLIC_HOSTNAME}" --routing-suffix="${OC_PUBLIC_IP}.nip.io"
+    wait_ocp
+}
+
+deploy_che_to_ocp() {
+    #Repull init image only if DEFAULT_IMAGE_PULL_POLICY is set to Always
+    if [ $DEFAULT_IMAGE_PULL_POLICY == "Always" ]; then
+        docker pull "$IMAGE_INIT"
+    fi
+    docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:nightly destroy --quiet --skip:pull --skip:nightly
+    docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:nightly config --skip:pull --skip:nightly
+    cd "${CONFIG_DIR}/instance/config/openshift/scripts/"
```

Comment:
please, add an ability to use the local script for easier development

Reply:
using local script is not right way for development as it is using files that should be generated. For development you have to build init image and set it as custom image for CLI.